projects/kit/components/input-phone-international/input-phone-international.component.ts

```diff
@@ -0,0 +1,246 @@
+import type {TuiFocusableElementAccessor} from '@taiga-ui/cdk';
+import {
+    AbstractTuiControl,
+    CHAR_PLUS,
+    tuiAsControl,
+    tuiAsFocusableItemAccessor,
+    tuiIsInputEvent,
+    tuiIsNativeFocused,
+    TuiLetDirective,
+} from '@taiga-ui/cdk';
+import {
+    TuiDataList,
+    TuiDropdownModule,
+    tuiDropdownOptionsProvider,
+    TuiFlagPipe,
+    TuiGroupDirective,
+    TuiHint,
+    TuiSelectDirective,
+    TuiTextfield,
+    TuiTextfieldOptionsDirective,
+    tuiTextfieldOptionsProvider,
+} from '@taiga-ui/core';
+import type {TuiCountryIsoCode} from '@taiga-ui/i18n';
+import {TuiChevronDirective} from '@taiga-ui/kit/directives';
+import {TUI_COUNTRIES} from '@taiga-ui/kit/tokens';
+import {PolymorpheusModule} from '@tinkoff/ng-polymorpheus';
+import {validatePhoneNumberLength} from 'libphonenumber-js';
+import type {MetadataJson} from 'libphonenumber-js/core';
+import {getCountryCallingCode} from 'libphonenumber-js/core';
+import {BehaviorSubject, combineLatest, distinctUntilChanged, map, skip} from 'rxjs';
+
+import {TuiGetCountryCallingCodePipe} from './get-country-calling-code.pipe';
+import {TUI_INPUT_PHONE_INTERNATIONAL_OPTIONS} from './input-phone-international.options';
+
+const NOT_FORM_CONTROL_SYMBOLS = /[^+\d]/g;
+
+@Component({
+    standalone: true,
+    selector: 'tui-input-phone-international',
+    imports: [
+        CommonModule,
+        FormsModule,
+        MaskitoDirective,
+        PolymorpheusModule,
+        TuiChevronDirective,
+        TuiDataList,
+        TuiDropdownModule,
+        TuiFlagPipe,
+        TuiGetCountryCallingCodePipe,
+        TuiGroupDirective,
+        TuiHint,
+        TuiLetDirective,
+        TuiTextfield,
+    ],
+    templateUrl: './input-phone-international.template.html',
+    styleUrls: ['./input-phone-international.style.less'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [
+        tuiAsFocusableItemAccessor(TuiInputPhoneInternationalComponent),
+        tuiAsControl(TuiInputPhoneInternationalComponent),
+        tuiTextfieldOptionsProvider({cleaner: false}),
+    ],
+    host: {
+        '[attr.data-size]': 'textfieldOptions?.size',
+    },
+    viewProviders: [
+        tuiDropdownOptionsProvider({
+            limitWidth: 'fixed',
+            align: 'right',
+        }),
+    ],
+})
+export class TuiInputPhoneInternationalComponent
+    extends AbstractTuiControl<string>
+    implements TuiFocusableElementAccessor
+{
+    /* eslint-disable @typescript-eslint/member-ordering */
+    protected readonly options = inject(TUI_INPUT_PHONE_INTERNATIONAL_OPTIONS);
+    protected readonly countryIsoCode$ = new BehaviorSubject(this.options.countryIsoCode);
+    protected readonly mask$ = new BehaviorSubject<MaskitoOptions | null>(null);
+    protected readonly countriesNames$ = inject(TUI_COUNTRIES);
+    protected readonly textfieldOptions = inject(TuiTextfieldOptionsDirective, {
```

Comment:
I think there's a token `TUI_TEXTFIELD_OPTIONS` instead of the optional directive?

Reply:
Fixed in https://github.com/taiga-family/taiga-ui/pull/7580/commits/814e19734dec70b4e86c6214d264dd9bbf580b79